src/core/Target.mli

```diff
@@ -0,0 +1,146 @@
+          selecting the relevant set of rules. *)
+  lockfile : lockfile option;
+      (** Optional lockfile associated with this target.
+
+          The association is namely that this target has its dependencies
+          specified by the given lockfile. Core doesn't need to worry about
+          determining these associations; rather, the target selection and
+          generation process must resolve these connections as part of
+          generating code targets. *)
+}
+[@@deriving show]
+(** A "normal" semgrep target, comprising source code (or, for
+   regex/generic, arbitrary text data) to be executed. See also {!Xtarget.t},
+   an augmented version which also has the contents. *)
+
+and lockfile = {
+  path : path;
+  kind : Lockfile_kind.t;
+      (** The type of lockfile this is. Analogous to analyzer for a source code
+          target. *)
+  manifest : manifest option;
+      (** Optionally, a manifest file associated with this lockfile. *)
+}
+[@@deriving show]
+(** A lockfile to be used during matching. See also {!Lockfile_xtarget.t}, an
+    augmented version with the contents of the lockfile. *)
+
+and manifest = {
+  path : path;
+  kind : Manifest_kind.t;
+      (** The type of manifest this is. Analogous to analyzer for a source code
+        target. *)
+}
+[@@deriving show]
+(** A manifest file to be used during matching. See also
+    {!Lockfile_xtarget.manifest}, which also has the contents. *)
+
+and path = {
+  origin : Origin.t;
+      (** The origin of the data as is relevant to the user. This could be,
+          e.g., a relative (from the project root) path to a file, a git object
+          and associated information, or anything else a Origin.t can
+          designate.
+
+          This should be used when reporting a location to the user. *)
+  internal_path_to_content : Fpath.t;
+      (** The path to a file which contains the data to be scanned. This could
+          be the same as the origin, if the origin is a path to a regular file
+          (or an absolute path to the same), or it could be a tempfile. This
+          should be used to obtain the contents of the target, but not for
+          reporting to the user, other than possibly for debugging purposes. *)
+}
+[@@deriving show]
+(** Information about where a target from for both the purpose of
+   {ul
+    {- informing the user: [origin]}
+    {- obtaining the contents: [internal_path_to_content]}
+  } *)
+
+(* deriving eq appears to not work on mutual definitions with and. See also
+   <https://github.com/ocaml-ppx/ppx_deriving/issues/272>, for deriving make,
+   which may be relevant. Therefore, we just define it here. *)
+val equal_path : path -> path -> bool
+
+val mk_code :
+  ?lockfile:lockfile ->
+  Xlang.t ->
+  Semgrep_output_v1_t.product list ->
+  Origin.t ->
+  code
+(** [mk_code analyzer products origin] is the target
+      location for a source code target originating from [origin] to be
```

Comment:
then maybe call it mk_source_code_target

Reply:
I renamed this in e18e0a6b4e5b7776a80ac46ff158b1490806f156 to `mk_regular`. In general I think this should parallel the name of the constructor for `Target.t`. If you disagree please let me know, but I hesitate to break that correspondence without a compelling reason.